File: src/main/java/com/senderman/stickersorterbot/bot/command/CommandExecutor.java
package com.senderman.stickersorterbot.bot.command;

import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;


/**
 * Interface which represents executor of commands given to bot by user
 */

public interface CommandExecutor {

    /**
     * The command itself. For example, "/help". It takes value from @Component annotation, by default
     */
    default String getCommand() {
        return getMyCommand(this);
    }

    /**
     * Description of the command
     */
    String getDescription();


    /**
     * Indicates if the command should be displayed in help message
     */
    default boolean showInHelp() {
        return true;
    }


    /**
     * Executes the logic of the given command.
     *
     * @param message user's message with command
     */
    void execute(Message message);


    private String getMyCommand(CommandExecutor executor) {
        return executor.getClass().getAnnotation(Component.class).value();
    }

}

File: src/main/kotlin/com/senderman/stickersorterbot/bot/CommandExecutor.kt
package com.senderman.stickersorterbot.bot

import org.springframework.stereotype.Component
import org.telegram.telegrambots.meta.api.objects.Message
import kotlin.reflect.full.findAnnotation

/**
 * Interface which represents executor of commands given to bot by user
 */
interface CommandExecutor {

    /**
     * The command itself. For example, "/help"
     */
    val command: String

    /**
     * Description of command
     */
    val desc: String

    /**
     * Indicates if the command should be displayed in help message
     */
    val showInHelp: Boolean
        get() = false


    /**
     * Executes the logic of the given command.
     * @param message user's message with command
     */
    fun execute(message: Message)

}

fun CommandExecutor.getMyCommand(): String = this::class.findAnnotation<Component>()!!.value